src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -385,6 +384,11 @@ internal void SerializeEncoding(SerializationInfo info, StreamingContext context
             }
         }
 
+        // On Desktop, encoding instances that aren't cached in a static field are cached in
+        // a hash table by codepage.
+        private static volatile Hashtable encodings;
```

Comment:
Also you should consider removing the `volatile` modifier, and instead using `Interlocked.CompareExchange` to initialize it. I believe the `lock` statement is still needed, however, since a regular Dictionary does not support multiple writers (so concurrent Adds may fail).